updates/upgrader.go

```diff
@@ -61,24 +61,35 @@ func upgrader(_ context.Context, _ interface{}) error {
 		log.Warningf("updates: failed to upgrade portmaster-start: %s", err)
 	}
 
-	if UpgradeCore {
+	binName := strings.TrimSuffix(
+		filepath.Base(os.Args[0]),
+		".exe",
+	)
+	switch binName {
+	case "portmaster-core":
```

Comment:
That check is likely to fail if the filenames are named differently. Also, at least on my machine the portmaster binary (`os.Args[0]`, `ps -ejh | grep portmaster`) is `/var/lib/portmaster/updates/linux_amd64/core/portmaster-core_v0-5-0` so the check actually fails already.